static/js/settings_panel_menu.js

```diff
@@ -1,14 +1,26 @@
+exports.mobile_deactivate_section = function () {
+    const $settings_overlay_container = $("#settings_overlay_container");
+    $settings_overlay_container.find(".right").removeClass("show");
+    $settings_overlay_container.find(".settings-header.mobile").removeClass("slide-left");
+};
+
 exports.make_menu = function (opts) {
     const main_elem = opts.main_elem;
     const hash_prefix = opts.hash_prefix;
     let curr_li = main_elem.children('li').eq(0);
 
     const self = {};
 
+    function two_column_mode() {
+        return $('#settings_overlay_container').css('--single-column') === undefined;
+    }
+
     self.show = function () {
         main_elem.show();
         const section = self.current_tab();
-        self.activate_section(section);
+        if (two_column_mode()) {
+            self.activate_section_or_default(section);
+        }
```

Comment:
Maybe worth adding a comment here noting that in the 1-column mode, we want to land at the menu, not the first settings tab, just to make the intent clear.